configstack/module.go

```diff
@@ -107,7 +108,7 @@ func resolveTerraformModule(terragruntConfigPath string, terragruntOptions *opti
 	return &TerraformModule{Path: modulePath, Config: *terragruntConfig, TerragruntOptions: opts}, nil
 }
 
-var moduleUrlRegexp = regexp.MustCompile(`.+//(.+?)(?:$|\?.+$)`)
+var moduleUrlRegexp = regexp.MustCompile(`(?:.+/)(.+?)(?:\?.+|\.)|(?:.+/)(.+)`)
```

Comment:
This regex is hard to read and understand. I stared at it for a long time and can't fully convince myself it'll work in all cases.

Perhaps we can simplify the regex portion by pre-processing the source URL a bit:

1. Strip of the query string (anything after and including a `?`)
1. Strip off the repo name (anything after and including a `.`)

Now we simply take the portion after the last slash.

Reply:
I like the query string removal idea. This avoids having to deal with slashes in the ref parameter. We can probably just use one of the split functions for that, no need for regex.

I have tested the regex against a number of cases, but I agree simpler == better. I'll look into implementing your proposals.